assembly/src/assembler/mod.rs

```diff
@@ -310,7 +318,7 @@ impl Assembler {
     /// # Errors
     ///
     /// Returns an error if parsing or compilation of the specified program fails.
-    pub fn assemble(&mut self, source: impl Compile) -> Result<Program, Report> {
+    pub fn assemble(self, source: impl Compile) -> Result<MastForest, Report> {
```

Comment:
nit: the comment on line 315 should be updated as we return `MastForest` rather than `Program` now.

Reply:
Fixed; moved the comment to the new `assemble_program()`.